src/firewall.c

```diff
@@ -89,29 +89,40 @@ unsigned int FW_MARK_MASK;             /**< @brief Iptables mask: bitwise or of
 char *
 arp_get(const char req_ip[])
 {
-	FILE *proc;
-	char ip[INET6_ADDRSTRLEN];
-	char mac[18];
+	int type = 0;
+	int flags = 0;
+	char ip[16] = {0};
+	char mac[18] = {0};
+	char mask[16] = {0};
+	char line[128] = {0};
+	char dev[128] = {0};
 	char *reply = NULL;
+	FILE *proc_arp = NULL;
 
-	if (!(proc = fopen("/proc/net/arp", "r"))) {
+	if (!(proc_arp = fopen("/proc/net/arp", "r"))) {
 		return NULL;
 	}
 
 	/* Skip first line */
-	while (!feof(proc) && fgetc(proc) != '\n');
+	fgets(line, sizeof(line), proc_arp);
 
 	/* Find ip, copy mac in reply */
-	reply = NULL;
-	while (!feof(proc) && (fscanf(proc, " %15[0-9.] %*s %*s %17[A-Fa-f0-9:] %*s %*s", ip, mac) == 2)) {
-		if (strcmp(ip, req_ip) == 0) {
+	while (fgets(line, sizeof(line), proc_arp))
+	{
+		if (sscanf(line, "%s 0x%x 0x%x %s %s %s\n",
```

Comment:
I don't think Nodogsplash gets the mac address from /proc/net/arp. 
Here is an example of syslog as my laptop connects to the wifi, re-negotiates DHCP and has a token allocated by Nodogsplash then a few seconds later I log in. After this /proc/net/arp is still empty.

Mon Jul 25 06:54:31 2016 daemon.info hostapd: wlan0-1: STA 74:de:2b:b1:70:33 IEEE 802.11: authenticated
Mon Jul 25 06:54:31 2016 daemon.info hostapd: wlan0-1: STA 74:de:2b:b1:70:33 IEEE 802.11: associated (aid 1)
Mon Jul 25 06:54:31 2016 daemon.info dnsmasq-dhcp[1016]: DHCPREQUEST(br-lan) 10.168.10.163 74:de:2b:b1:70:33 
Mon Jul 25 06:54:31 2016 daemon.info dnsmasq-dhcp[1016]: DHCPACK(br-lan) 10.168.10.163 74:de:2b:b1:70:33 linux-8nro
Mon Jul 25 06:54:34 2016 daemon.info dnsmasq-dhcp[1016]: DHCPREQUEST(br-lan) 10.168.10.163 74:de:2b:b1:70:33 
Mon Jul 25 06:54:34 2016 daemon.info dnsmasq-dhcp[1016]: DHCPACK(br-lan) 10.168.10.163 74:de:2b:b1:70:33 linux-8nro
Mon Jul 25 06:57:49 2016 daemon.notice nodogsplash[1092]: Adding 10.168.10.163 74:de:2b:b1:70:33 token e6a826a1 to client list
Mon Jul 25 06:58:03 2016 daemon.notice nodogsplash[1092]: Authenticating 10.168.10.163 74:de:2b:b1:70:33